### DISCUSSION TITLE
Spleen segmentation example -- where is softmax applied at validation?

### DISCUSSION THREAD

**Discussion by philnovv** [Project-MONAI/tutorials#1425] [General]
Hi, in the spleen segmentation example here:
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb

I understand that the output of the model has not been activated by a softmax. Hence, we pass softmax=True to the loss function. But during the validation, **when does softmax get applied**?

We have:
val_outputs = sliding_window_inference(val_inputs, roi_size, sw_batch_size, model) \
val_outputs = [post_pred(i) for i in decollate_batch(val_outputs)] 

but  post_pred does not apply softmax. What am I missing?

Contrast this to the following segmentation example notebook:
https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py

where post_pred (here "post_trans") contains the sigmoid activation. Can anyone clarify?

Cheers

**Comment by KumoLiu** [comment]
Duplication with https://github.com/Project-MONAI/MONAI/discussions/6619.
